notification/shoutrrr.go

```diff
@@ -10,6 +12,33 @@ import (
 	"github.com/flanksource/incident-commander/api"
 )
 
+// defaultSMTPPrefix indicates that the shoutrrr URL for smtp should use
+// the system's SMTP credentials.
+const defaultSMTPPrefix = "smtp://auto:auto@auto/"
+
+// setSystemSMTPCredential modifies the shoutrrrURL to use the system's SMTP credentials.
+func setSystemSMTPCredential(shoutrrrURL string) (string, error) {
+	prefix := fmt.Sprintf("smtp://%s:%s@%s:%s/",
+		url.QueryEscape(os.Getenv("SMTP_USER")),
+		url.QueryEscape(os.Getenv("SMTP_PASSWORD")),
+		os.Getenv("SMTP_HOST"),
+		os.Getenv("SMTP_PORT"),
+	)
+	shoutrrrURL = strings.ReplaceAll(shoutrrrURL, defaultSMTPPrefix, prefix)
+
+	parsedURL, err := url.Parse(shoutrrrURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse shoutrrr URL: %w", err)
+	}
+
+	query := parsedURL.Query()
+	query.Set("FromAddress", os.Getenv("SMTP_USER"))
```

Comment:
I think we are also missing an SMTP_FROM - note the user and from is not necessarily the same (if from not specified then use SMTP_USER)

Reply:
We seem to be taking SMTP from as a flag. Will use that.